libs/books/feature/src/lib/book-search/book-search.component.ts

```diff
@@ -48,7 +50,18 @@ export class BookSearchComponent implements OnInit,OnDestroy {
   addBookToReadingList(book: Book) {
     this.store.dispatch(addToReadingList({ book }));
   }
-
+  
+  onChanges(){
+    this.searchForm.get('term').valueChanges.pipe(
```

Comment:
The subscription  needs to be cleaned up.

Also, don't use `switchMap` or other mapping operators for anything other than data mapping. In fact, `store.dispatch` returns void so there is nothing to map from it!

```
// ...
    this.searchForm.get('term').valueChanges.pipe(
      filter( data => data.trim().length > 0 )
      debounceTime(500)
    ).subscribe(term =>{
      this.store.dispatch(searchBooks({ term }))
    })
```